airgun/widgets.py

```diff
@@ -1322,9 +1433,40 @@ def read(self):
         if not self.has_rows:
             self.logger.debug('Table {} is empty'.format(self.locator))
             return []
-
+        if self.pagination.is_displayed:
+            return self._read_all()
         return super(SatTable, self).read()
 
+    def _read_all(self):
+        """Return all available table values with using pagination navigation."""
+        table_rows = []
+        page_number = 1
+        if self.pagination.current_page != page_number:
+            # goto first page
+            self.pagination.first_page()
+            wait_for(
+                lambda: self.pagination.current_page == page_number,
+                timeout=30,
+                delay=1,
+                logger=self.logger
+            )
+        while True:
```

Comment:
I'd rather prefer smith like
```python
for index in range(self.pagination.total_pages):
    page_table_rows = super().read()
    table_rows.extend(page_table_rows)
    if index + 1 != self.pagination.total_pages:
        self.pagination.next_page()
        wait_for(
            lambda: self.pagination.current_page == index + 1,
            timeout=30,
            delay=1,
            logger=self.logger
        
```
or smth similar (e.g. with `break`) and tried to avoid `while True:` at any cost, especially as this case is resolvable via `for` loop.

Reply:
after conversation came to consensus :)